argmin-math/src/lib.rs

```diff
@@ -275,3 +279,102 @@ pub trait ArgminMinMax {
     /// Select piecewise maximum
     fn max(x: &Self, y: &Self) -> Self;
 }
+
+/// Defines associated types
+pub trait ArgminTransition {
```

Comment:
It is not entirely clear to me what this does and what it is needed for. Could you elaborate a bit, please?

Reply:
I need to some mechanism that will return me a type of 2-D array in exchange for 1-D array. Take a look at [this line of code](https://github.com/argmin-rs/argmin/blob/e4bb91a84fd3f2d5173524bf8b3916f06734a046/argmin/src/solver/cma_es/mod.rs#L135). To create an instance of 2-D identity matrix I need to know a type of the matrix. The problem is all my parameters are either scalars or 1-D vectors. This class helps me to transition from 1-D to 2-D array.

Reply:
Thanks for the explanation, makes sense!